client/src/util/camera.ts

```diff
@@ -50,43 +52,114 @@ export class Camera {
     return this.viewMatrix[0];
   }
 
-  pan(dx: number, dy: number): void {
+  /**
+   * Pans the camera by the specified amount in the x and y directions.
+   *
+   * @param dx - The amount to pan in the x direction.
+   * @param dy - The amount to pan in the y direction.
+   * @param openseadragon - The OpenSeadragon viewer instance (optional).
+   * @param offsetX - The x offset of the pan operation (default: 0).
+   * @param offsetY - The y offset of the pan operation (default: 0).
+   * @param prevOffsetX - The previous x offset of the pan operation (default: 0).
+   * @param prevOffsetY - The previous y offset of the pan operation (default: 0).
+   */
+  pan({
+    dx,
+    dy,
+    openseadragon,
+    offsetX = 0,
+    offsetY = 0,
+    prevOffsetX = 0,
+    prevOffsetY = 0,
+  }: {
+    dx: number;
+    dy: number;
+    openseadragon: Viewer | null;
+    offsetX?: number;
+    offsetY?: number;
+    prevOffsetX?: number;
+    prevOffsetY?: number;
+  }): void {
     const m = this.viewMatrix;
-    const dyRange: [number, number] = [
-      -panBound - (m[7] + 1) / m[4],
-      panBound - (m[7] - 1) / m[4],
-    ];
-    const dxRange: [number, number] = [
-      -panBound - (m[6] + 1) / m[0],
-      panBound - (m[6] - 1) / m[0],
-    ];
+
+    const dyRange: [number, number] = [-(m[7] + 1) / m[4], -(m[7] - 1) / m[4]];
+    const dxRange: [number, number] = [-(m[6] + 1) / m[0], -(m[6] - 1) / m[0]];
 
     const dxClamped = clamp(dx, dxRange);
     const dyClamped = clamp(dy, dyRange);
-    if (Math.abs(dxClamped) <= EPSILON && Math.abs(dyClamped) <= EPSILON)
+
+    if (Math.abs(dxClamped) <= EPSILON && Math.abs(dyClamped) <= EPSILON) {
       return;
+    }
 
-    mat3.translate(m, m, [dxClamped, dyClamped]);
+    mat3.translate(m, m, [dx, dy]);
```

Comment:
not using the clamped values to translate to ensure the embedding is in sync with openseadragon